package/contents/ui/lib/WindowLayout.qml

```diff
@@ -13,26 +13,48 @@ PlasmaComponents.Button {
     implicitWidth: 160*1.2 * PlasmaCore.Units.devicePixelRatio
     implicitHeight: 90*1.2 * PlasmaCore.Units.devicePixelRatio
 
+    property var main
     property var windows
+    property var screen
     property var clickedWindows: []
 
-    function tileWindow(window, x, y, width, height) {
-        if (!window.normalWindow) return;
-        if (rememberWindowGeometries && !oldWindowGemoetries.has(window)) oldWindowGemoetries.set(window, [window.geometry.width, window.geometry.height]);
-
-        let screen = workspace.clientArea(KWin.MaximizeArea, workspace.activeScreen, window.desktop);
+    function tileWindow(client, window, root) {
+        var screen = root.screen;
+        if (screen == undefined) { console.log("screen not defined"); return; }
+        if (!client.normalWindow) return;
+        if (root.main.rememberWindowGeometries && !root.main.oldWindowGemoetries.has(client)) root.main.oldWindowGemoetries.set(client, [client.geometry.width, client.geometry.height]);
 
         let xMult = screen.width / 12.0;
         let yMult = screen.height / 12.0;
+        let x = 0.0;
+        let y = 0.0;
+        let width = 0.0;
+        let height = 0.0;
+
+        if (window.hasOwnProperty("rawX")) {
+            x = window.rawX;
+        } else {
+            x = Math.round(window.x * xMult) + tileGap;
+        }
+        if (window.hasOwnProperty("rawY")) {
+            y = window.rawY;
+        } else {
+            y = Math.round(window.y * yMult) + tileGap;
+        }
+        if (window.hasOwnProperty("rawWidth")) {
+            width = window.rawWidth;
+        } else {
+            width = Math.round(window.width * xMult) - 2*tileGap;
+        }
+        if (window.hasOwnProperty("rawWidth")) {
+            height = window.rawHeight;
+        } else {
+            height = Math.round(window.height * yMult) - 2*tileGap;
+        }
```

Comment:
Something like
```js
x = window.rawX ? window.rawX : Math.round(window.x * xMult) + tileGap;
```
or (if nullish coalescing is supported in QML JS, not sure about that)
```js
x = window.rawX ?? Math.round(window.x * xMult) + tileGap;
```
would be more concise here.

Reply:
Done.